core/message_pool.go

```diff
@@ -250,10 +139,24 @@ func (pool *MessagePool) LargestNonce(address address.Address) (largest uint64,
 	return
 }
 
+// PendingBefore returns the CIDs of messages added with height less than `minimumHeight`.
+func (pool *MessagePool) PendingBefore(minimumHeight uint64) []cid.Cid {
+	pool.lk.RLock()
+	defer pool.lk.RUnlock()
+
+	var cids []cid.Cid
+	for c, msg := range pool.pending {
+		if msg.addedAt < minimumHeight {
+			cids = append(cids, c)
+		}
+	}
+	return cids
```

Comment:
Not something we need to address in this PR, but we could get a big performance improvement here by sorting pending upon insertion and exiting early here. It might be a good idea to add an icebox task for optimizing this.